packages/grid/x-data-grid/src/models/gridSlotsComponentsProps.ts

```diff
@@ -1,28 +1,83 @@
+import * as React from 'react';
+import { CheckboxProps } from '@mui/material/Checkbox';
+import { TextFieldProps } from '@mui/material/TextField';
+import { FormControlProps } from '@mui/material/FormControl';
+import { SelectProps } from '@mui/material/Select';
+import { SwitchProps } from '@mui/material/Switch';
+import { ButtonProps } from '@mui/material/Button';
+import { IconButtonProps } from '@mui/material/IconButton';
+import { TooltipProps } from '@mui/material/Tooltip';
+import { PopperProps } from '@mui/material/Popper';
+import { TablePaginationProps } from '@mui/material/TablePagination';
+import { GridToolbarProps } from '../components/toolbar/GridToolbar';
+import { ColumnHeaderFilterIconButtonProps } from '../components/columnHeaders/GridColumnHeaderFilterIconButton';
+import { GridColumnMenuProps } from '../components/menu/columnMenu/GridColumnMenuProps';
+import { GridColumnsPanelProps } from '../components/panel/GridColumnsPanel';
+import { GridFilterPanelProps } from '../components/panel/filterPanel/GridFilterPanel';
+import { GridFooterContainerProps } from '../components/containers/GridFooterContainer';
+import { GridOverlayProps } from '../components/containers/GridOverlay';
+import { GridPanelProps } from '../components/panel/GridPanel';
+import { GridRowProps } from '../components/GridRowProps';
+import { GridCellProps } from '../components/cell/GridCellProps';
+
+// Overrides for module augmentation
+export interface BaseCheckboxPropsOverrides {}
```

Comment:
Alternatively, we could expose a single interface for overrides with the same keys as `GridSlotsComponentsProps`:

```tsx
type SlotPropsOverrides = {
  [slotName in keyof GridSlotsComponentsProps]: {};
};

// Overrides for module augmentation
export interface GridSlotPropsOverrides extends SlotPropsOverrides {}

export interface GridSlotsComponentsProps {
  baseCheckbox?: SlotProps<CheckboxProps, GridSlotPropsOverrides['baseCheckbox']>;
  baseTextField?: SlotProps<TextFieldProps, GridSlotPropsOverrides['baseTextField']>;
```

Then you could augment it like this:
```tsx
declare module '@mui/x-data-grid' {
  interface GridSlotPropsOverrides {
    baseCheckbox: {
      customProp: string;
    }
    baseTextField: {
      customProp: number;
    }
  }
}
```
as opposed to:
```tsx
declare module '@mui/x-data-grid' {
  interface BaseCheckboxPropsOverrides {
    customProp: string;
  }
  
  interface BaseTextFieldPropsOverrides {
    customProp: number;
  }
}
```

The main difference is DX, which in my opinion is slightly more intuitive when using a single interface - you don't have to "guess" the interface name and its shape is identical to the `slotProps` object shape.

Another small difference is a bit easier maintenance - no need to add a new overrides interface when the new slot is added - it would work automatically (because it uses mapped type).

What do you think?
cc @mui/x - this might be interesting for the eXplore team as well.

Reply:
I think the priority should be to do the same as the core which is currently discussing exactly the same topic.

Reply:
Is there an issue for it in the core repo?

Reply:
I found https://github.com/mui/material-ui/issues/34906

Reply:
Yes, I tried to follow the same approach as core (separate override each slot)

I really liked the idea though, thanks for the suggestion @cherniavskii, it could be much better `dx` wise. Not sure though if following the same approach as `core` approach makes more sense.

Reply:
I've added a comment in https://github.com/mui/material-ui/issues/34906, let's discuss it and see where it would bring us 🙂

Reply:
By the way if we go with the current approach, a section like this in demo will possibly reduce some friction.

### Custom slot props with Typescript

`--- other content ---`

The naming of overridable interfaces uses a pattern like this: 
<pre>
`${slotNameInPascalCase}PropsOverrides`
</pre>
For example, for `columnMenu` slot, the interface name would be `ColumnMenuPropsOverrides`.

This [file](https://github.com/mui/mui-x/blob/next/packages/grid/x-data-grid/src/models/gridSlotsComponentsProps.ts) lists all the interfaces for each slot which could be used for augmentation.

Reply:
Looks good to me, let's proceed with it!